src/server/qgshttprequesthandler.cpp

```diff
@@ -218,21 +218,10 @@ void QgsHttpRequestHandler::sendResponse()
   clearBody();
 }
 
-QByteArray QgsHttpRequestHandler::getResponse( const bool returnHeaders,
-    const bool returnBody )
+QPair<QByteArray, QByteArray> QgsHttpRequestHandler::getResponse()
 {
-  if ( ! returnHeaders )
-  {
-    return mResponseBody;
-  }
-  else if ( ! returnBody )
-  {
-    return mResponseHeader;
-  }
-  else
-  {
-    return mResponseHeader.append( mResponseBody );
-  }
+  QPair<QByteArray, QByteArray> response( mResponseHeader, mResponseBody );
+  return response;
```

Comment:
Hi @elpaso 

* Why not introducing a new type QgsHttpResponse instead of the QPair with header and body members? This will add the flexibility to add more information to it later (security context, timestamp, encryption information... I don't know the state and internals of the server, I just assume that some kind of requirement could arise sooner or earlier.

* Using a stackallocated `QByteArray` should be fine, since it's implicitly shared and can be copied with low overhead.

* Is it optimal to make both - header and body - a `QByteArray`? I guess for the body it's to be able to pass around blobs like images, but does that also apply for the header or could that be a (encoding aware) `QString` instead?